File: 02/CV02/Utorok/Aritmetika/aritmetika.py
def fact1(n):
    """
    nerekurzivny faktorial, cyklus for  
    """
    f=1
    for i in range(1,n):  # 1,2,.., n-1
        f *= i+1
    return f

def fact2(n):
    """
    rekurzivny faktorial, nepouziva priradenie
    """
    if n==1:
        return 1
    else:
        return n*fact2(n-1)
    
def times3(a,b):
    """
    rekurzivny faktorial, nepouziva priradenie, nepouziva nasobenie
    """
    if a == 0:
        return 0
    else:
        return times3(a-1,b)+b;

def fact3(n):
    if n==1:
        return 1
    else:
        return times3(n,fact3(n-1))
    
def times4(a,b):
    """
    rekurzivny faktorial, nepouziva priradenie, nepouziva nasobenie, nepouziva ziadne ciselne konstanty
    """
    if a-a == a:
        return a
    else:
        return times3(a-(a/a),b)+b;

def fact4(n):
    if n == n//n:
        return n
    else:
        return times4(n,fact4(n-(n//n)))

###############################################################
#  priklady obmedzeneho Pythona s jedinou globalnou premennou a
###############################################################

a = 0  # toto je ta globalna premenna, na to aby sme ju videli v lokalnom scope, musime napisat 'global a'


def sum2Nat():
    """
    sucet dvoch prirodzenych cisel s jednou globalnou premennou
    """
    global a
    a = int(input('zadaj prve cislo:'))
    print ('ich sucet je: ', str(a))

sum2Nat()
#sum2Int()
#sum3Int()
#max2Nat()
    
        
        

File: 02/CV02/Utorok/Aritmetika/aritmetika.go
package main

import (
	"fmt"
	"strconv"
)

/*
#####################################################################

# GO1 - Napíšte v jazyku Go program na výpočet faktorialu, ktorý
#   - nepoužíva príkaz priradenia,
#   - nepoužíva násobenie,
#   - nepoužíva žiadne číselne konštanty.
*/

func mult(a, b int) int {
	if a == a+b {
		return b
	} else {
		return mult(a, b-(b/b)) + a
	}
}

func fact(n int) int {
	if n == n+n {
		return ^n / ^n
	} else {
		return mult(n, fact(n-(n/n)))
	}
}

/*
###############################################################
#  obmedzené príklady s jedinou globalnou premennou a
###############################################################
*/

// toto je ta globalna premenna
var a int = 0

// toto nemeniť, len použiť
func input(prompt string) {
	var i string
	fmt.Println(prompt)
	fmt.Scan(&i)
	a, _ = strconv.Atoi(i)
}

// súčet dvoch prirodzených čísel s jednou globalnou premennou
func sum2Nat() {
	if a > 0 {
		a--
		sum2Nat()
		a++
	} else {
		input("druhe")
	}
}

// súčet dvoch celych čísel s jednou globalnou premennou
func sum2Int() {
	if a > 0 {
		a--
		sum2Int()
		a++
	} else if a < 0 {
		a++
		sum2Int()
		a--
	} else {
		input("druhe")
	}
}

// súčet troch celych čísel s jednou globalnou premennou
func sum3Int() {
	sum2Int()
	sum2Int()
}

// maximum dvoch prirodzenych čísel s jednou globalnou premennou
func dvaA() {
	if a == 0 {
		input("druhe")
		triB()
	} else {
		a--
		dvaA()
		a*=2
	}
}

func triB() {
	if a == 0 {
		a = 1
	} else {
		a--
		triB()
		a*=3
	}
}

func min() {
	if a % 6 != 0 {
		a = 0
	} else {
		a /= 6
		min()
		a++
	}
}

func main() {
	input("prve")
	dvaA()
	min()
	fmt.Printf("ich min je: %d\n", a)

}


func comb(n, k int64) int64 {
	if k == 0 || n == k {
		return 1
	} else {
		return comb(n-1, k-1) + comb(n-1, k)
	}
}
